cmd/dagger/clientgen.go

```diff
@@ -13,40 +13,41 @@ import (
 	"github.com/spf13/cobra"
 
 	"go.dagger.io/dagger/codegen/generator"
-	"go.dagger.io/dagger/engine"
+	"go.dagger.io/dagger/sdk/go/dagger"
 )
 
 var clientGenCmd = &cobra.Command{
-	Use: "client-gen",
-	Run: ClientGen,
+	Use:  "client-gen",
+	RunE: ClientGen,
 }
 
 func init() {
 	clientGenCmd.Flags().StringP("output", "o", "", "output file")
 	clientGenCmd.Flags().String("package", "", "package name")
 }
 
-func ClientGen(cmd *cobra.Command, args []string) {
-	startOpts := &engine.Config{
-		Workdir:     workdir,
-		ConfigPath:  configPath,
-		SkipInstall: true,
+func ClientGen(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
+	c, err := dagger.Connect(ctx,
+		dagger.WithWorkdir(workdir),
```

Comment:
What happens if this isn't passed in? Is a workdir not set?

Just curious because I think there are use cases for not having one at all, so omitting it could help avoid an expensive upload.

Reply:
I believe `cwd` is the workdir if none is specified

Reply:
Yes that's the current behavior. We can definitely make workdir optional too though (in general, probably too much of an effort to tie into this PR). I mentioned that too here: https://github.com/dagger/dagger/issues/3375